circuit/environment/src/helpers/assignment.rs

```diff
@@ -29,8 +29,14 @@ impl<F: PrimeField> From<&crate::Variable<F>> for AssignmentVariable<F> {
     fn from(variable: &crate::Variable<F>) -> Self {
         match variable {
             crate::Variable::Constant(value) => Self::Constant(**value),
-            crate::Variable::Public(index, _) => Self::Public(*index),
-            crate::Variable::Private(index, _) => Self::Private(*index),
+            crate::Variable::Public(index_value) => {
```

Comment:
Can you use this format?

```suggestion
            crate::Variable::Public((index, value)) => {
```

Reply:
sadly, it's not possible for `Rc` to be pattern-matched like that:
```
error[E0308]: mismatched types
  --> circuit/environment/src/helpers/assignment.rs:32:37
   |
30 |         match variable {
   |               -------- this expression has type `&variable::Variable<F>`
31 |             crate::Variable::Constant(value) => Self::Constant(**value),
32 |             crate::Variable::Public((index,value)) => {
   |                                     ^^^^^^^^^^^^^ expected `Rc<(u64, F)>`, found `(_, _)`
   |
   = note: expected struct `Rc<(u64, F)>`
               found tuple `(_, _)
```